deeprankcore/trainer.py

```diff
@@ -688,23 +759,25 @@ def save_model(self, filename='model.pth.tar'):
             "weight_decay": self.weight_decay,
             "subset": self.subset,
             "shuffle": self.shuffle,
-            "clustering_method": self.clustering_method
+            "clustering_method": self.clustering_method,
+            "node_features": self.node_features,
+            "edge_features": self.edge_features,
+            "features": self.features
         }
 
         torch.save(state, filename)
 
 
-def _divide_dataset(dataset: GraphDataset, splitsize: Union[float, int] = None):
+def _divide_dataset(dataset: Union[GraphDataset, GridDataset], splitsize: Union[float, int] = None) -> \
+        Union[Tuple[GraphDataset, GraphDataset], Tuple[GridDataset, GridDataset]]:
+
     """Divides the dataset into a training set and an evaluation set
 
     Args:
-        dataset (GraphDataset): input dataset to be split into training and validation data
-        
+        dataset: input dataset to be split into training and validation data
```

Comment:
```suggestion
        dataset (deeprank-core dataset object): input dataset to be split into training and validation data
```

Reply:
Changed this in commit 9ad784c964ef77876f71e39599c150dcace6bb22.